client/src/cordova/apple/xcode/ios/Outline/Classes/AppKitBridge/StatusItemController.swift

```diff
@@ -100,6 +121,52 @@ class StatusItemController: NSObject {
         NotificationCenter.default.post(name: Notification.Name("appQuit"), object: nil)
         NSApplication.shared.terminate(self)
     }
+    
+    @objc func toggleVpnConnection(_: AnyObject?) {
+        NSLog("[StatusItemController] Toggle VPN connection")
+        
+        // Check current VPN status
+        Task {
+            let managers = try? await NETunnelProviderManager.loadAllFromPreferences()
+            let hasProfile = managers?.count ?? 0 > 0
+            
+            if hasProfile {
+                // Get the first (and only) VPN profile
+                guard let manager = managers?.first else {
+                    NSLog("[StatusItemController] Failed to get VPN manager")
+                    return
+                }
+                
+                let isActive = manager.connection.status == .connected || 
```

Comment:
This is a race condition. The action should be based on what was showing on the menu, not the current status. You can't disconnect when the user clicked connect.

If it said connect, and it's already connected, do nothing. Same for the disconnect case.
Should only act if the desired state doesn't match the current state.